src/libcore/ptr/mod.rs

```diff
@@ -18,6 +18,10 @@
 //! * A [null] pointer is *never* valid, not even for accesses of [size zero][zst].
 //! * All pointers (except for the null pointer) are valid for all operations of
 //!   [size zero][zst].
+//! * For a pointer to be valid, it is necessary (but not always sufficient) that the pointer
+//!   be *dereferencable*: the memory range of the given size starting at the pointer must all be
```

Comment:
This is fairly abstract and packed -- can we give an example perhaps?

Reply:
There's not really a good way (that I can think of) to do that in the current format -- it would need basically a re-write of this section, I think. And the phrasing is similar to our existing docs for, e.g., `ptr::offset`.

Reply:
Ah; alright -- better to have some docs than none. r=me if you are happy with the current state.